src/bin/cargo/main.rs

```diff
@@ -168,7 +168,17 @@ fn execute_external_subcommand(config: &Config, cmd: &str, args: &[&str]) -> Cli
         None => {
             let suggestions = list_commands(config);
             let did_you_mean = closest_msg(cmd, suggestions.keys(), |c| c);
-            let err = anyhow::format_err!("no such subcommand: `{}`{}", cmd, did_you_mean);
+
+            let err = if cmd.contains('+') {
+                anyhow::format_err!("no such subcommand: `{}`{}\n\n\tCargo does not handle `+toolchain` directives itself.\n\tDid you mean to run `cargo` through `rustup` instead?", cmd, did_you_mean)
+            } else {
+                if did_you_mean.is_empty() {
+                    anyhow::format_err!("no such subcommand: `{}`\n\n\tA similar Cargo subcommand could not be found\n\tView a list of installed Cargo subcommands using `cargo --list`", cmd)
```

Comment:
I personally find the phrase "A similar Cargo subcommand could not be found" superfluous.  The more text that exists, the more likely the user is going to miss what matters

Reply:
Yes I can see this happening. I originally was just thinking of adding **View a list of installed Cargo subcommands using `cargo --list`**, but it is apparent that I went down the more *verbose* route.